skfda/ml/regression/_linear_regression.py

```diff
@@ -49,10 +50,44 @@
 ]
 
 
+class DataFrameMixin:
+    """Mixin class to add DataFrame funcionality."""
+
+    def dataframe_conversion(self, X: pd.DataFrame) -> List:
+        """Convert DataFrames to a list with two elements: first of all, a list with mv
+        covariates and the second, a FDataBasis object with functional data.
+
+        Args:
+            - X: pandas DataFrame to convert.
+
+        Returns:
+            - list with two elements: first of all, a list with mv
+            covariates and the second, a FDataBasis object with functional data.
+        """
+        fdb = FDataBasis.concatenate(*X.iloc[:, 1].tolist())
```

Comment:
I think we want to convert to a list with one variable (multivariate or functional, whatever the case) per column. Don't try to concatenate things or apply fancy logic.